.github/workflows/ci.yml

```diff
@@ -0,0 +1,31 @@
+name: CI
+on: [push, pull_request]
+jobs:
+  build:
+    strategy:
+      matrix:
+        ruby: [2.3, 2.4, 2.5, 2.6, 2.7, jruby]
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+        exclude:
+          # eregon/use-ruby-action doesn't support JRuby on Windows
+          - ruby: jruby
+            platform: windows-latest
+          # jaro_winkler (rubocop dependency) fails to compile
+          - ruby: 2.3
+            platform: windows-latest
+      # TODO: remove this before merging
+      fail-fast: false
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Ruby
+        uses: eregon/use-ruby-action@master
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - name: Build
+        run: |
+          gem install bundler
+          bundle install --jobs 4 --retry 3
+      - name: Test
+        run: bundle exec ruby -w $(bundle exec ruby -e "print Gem.bin_path 'rake', 'rake'") spec
```

Comment:
(I wish rake honored the -w flag).

Reply:
Normally, setting RUBYOPT=-w would work, except it breaks the CLI tests. That's why I don't use it.

Reply:
I'd say these are of equal complexity, but as you wish.

Reply:
It's not just about complexity. It's the right location from which to load the bin script because it's the real bin script after gem install. The one inside the gem is not guaranteed to be a full bin script (as the gem may do processing on it during packaging).